webapp/lib/view.dart

```diff
@@ -1019,3 +1024,41 @@ class UrlView {
     return false;
   }
 }
+
+enum SnackbarNotificationType {
+  info,
+  success,
+  warning,
+  error
+}
+
+class SnackbarView {
+  DivElement snackbarElement;
+  DivElement _contents;
+
+  SnackbarView() {
+    snackbarElement = new DivElement()
+      ..id = 'snackbar'
+      ..classes.add('hidden')
+      ..title = 'Click to close notification.'
+      ..onClick.listen((_) => hideSnackbar());
+
+    _contents = new DivElement()
+      ..classes.add('contents');
+    snackbarElement.append(_contents);
+  }
+
+  showSnackbar(String message, SnackbarNotificationType type) {
+    _contents.text = message;
+    snackbarElement.classes.remove('hidden');
+    snackbarElement.setAttribute('type', type.toString().replaceAll('SnackbarNotificationType.', ''));
+    new Timer(new Duration(seconds: 3), () => hideSnackbar());
+  }
+
+  hideSnackbar() {
+    snackbarElement.classes.add('hidden');
+    snackbarElement.attributes.remove('type');
+    // Remove the contents after the animation ends - the duration here must match the animation lenth
```

Comment:
nit: s/lenth/length

Reply:
Could you make it explicit that the thing it has to match is defined in snackbar.css?

Reply:
Done, thanks!